specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2024-09-01-preview/redisenterprise.json

```diff
@@ -0,0 +1,2711 @@
+          }
+        },
+        "description": "Checks that the Redis Enterprise cache name is valid and is not already in use.",
+        "parameters": [
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CheckNameAvailabilityParameters"
+            },
+            "description": "Parameters supplied to the CheckNameAvailability Redis operation. The only supported resource type is 'Microsoft.Cache/redisenterprise'"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Name is available"
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CheckNameAvailabilityParameters": {
+      "description": "Parameters body to pass for resource name availability check.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "type": "string",
+          "description": "Resource type. The only legal value of this property for checking redis enterprise cache name availability is 'Microsoft.Cache/redisenterprise'."
+        }
+      },
+      "required": [
+        "name",
+        "type"
+      ]
+    },
+    "Sku": {
+      "description": "SKU parameters supplied to the create Redis Enterprise cluster operation.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The level of Redis Enterprise cluster to deploy. Possible values: ('GeneralPurpose_G5', 'MemoryOptimized_M20', etc.). For more information on SKUs see the latest pricing documentation. Note that additional SKUs may become supported in the future.",
+          "enum": [
+            "Enterprise_E1",
+            "Enterprise_E5",
+            "Enterprise_E10",
+            "Enterprise_E20",
+            "Enterprise_E50",
+            "Enterprise_E100",
+            "Enterprise_E200",
+            "Enterprise_E400",
+            "EnterpriseFlash_F300",
+            "EnterpriseFlash_F700",
+            "EnterpriseFlash_F1500",
+            "GeneralPurpose_G0",
```

Comment:
We need to confirm the SKU name here.

Reply:
Will ask Jordan to confirm here.

Reply:
![image](https://github.com/user-attachments/assets/3ff805a5-bd40-4a67-8564-e363a4979a42)

Updated based Jordan's comment